src/ide/code_actions/suggest_similar_method.rs

```diff
@@ -0,0 +1,82 @@
+use crate::lang::analysis_context::AnalysisContext;
+use crate::lang::db::AnalysisDatabase;
+use crate::lang::lsp::ToLsp;
+use crate::lang::methods::find_methods_for_type;
+use crate::lang::text_matching::text_matches;
+use cairo_lang_defs::ids::NamedLanguageElementLongId;
+use cairo_lang_semantic::db::SemanticGroup;
+use cairo_lang_semantic::items::function_with_body::SemanticExprLookup;
+use cairo_lang_semantic::lookup_item::LookupItemEx;
+use cairo_lang_syntax::node::{TypedStablePtr, TypedSyntaxNode, ast};
+use cairo_lang_utils::Upcast;
+use lsp_types::{CodeAction, CodeActionKind, Range, TextEdit, Url, WorkspaceEdit};
+use std::collections::HashMap;
+
+/// Create a Quick Fix code action to substitute this call with a similar one.
+pub fn suggest_similar_method<'db>(
+    db: &'db AnalysisDatabase,
+    ctx: &AnalysisContext<'db>,
+    uri: Url,
+) -> Option<Vec<CodeAction>> {
+    let expr_binary = ctx.node.ancestor_of_type::<ast::ExprBinary>(db)?;
+    let lhs_stable_ptr = expr_binary.lhs(db).stable_ptr(db);
+    // Get its semantic model.
+    let function_with_body = ctx.lookup_item_id?.function_with_body()?;
+
+    let expr_id = db
+        .lookup_expr_by_ptr(function_with_body, lhs_stable_ptr)
+        .inspect_err(|err| eprintln!("{:?}", err))
+        .ok()?;
+
+    let semantic_db: &dyn SemanticGroup = db.upcast();
+    let ty = semantic_db.expr_semantic(function_with_body, expr_id).ty();
+
+    if ty.is_missing(db) {
+        return None;
+    }
+    eprintln!("ty: {:?}", ty);
+    let method_names =
+        find_methods_for_type(db, &mut ctx.resolver(db), ty, lhs_stable_ptr.untyped());
+
+    let matching_methods: Vec<String> = method_names
+        .into_iter()
+        .filter_map(|function_id| {
+            let fn_name = function_id.long(db).name(db);
+            let name_matches = text_matches(fn_name, ctx.node.get_text(db));
+            if name_matches {
+                return Some(fn_name);
+            }
+            None
+        })
+        .map(|name| name.to_string())
+        .collect();
+    let only_one_method = matching_methods.len() == 1;
+    let bad_method_name_span =
+        ctx.node.span(db).position_in_file(db, ctx.module_file_id.file_id(db).ok()?)?;
+
+    let code_actions = matching_methods
+        .into_iter()
+        .map(|method_name| {
+            let edit = TextEdit {
+                range: Range {
+                    start: bad_method_name_span.start.to_lsp(),
+                    end: bad_method_name_span.end.to_lsp(),
+                },
+                new_text: method_name.to_string(),
+            };
+            CodeAction {
+                title: format!("Use {method_name} instead"),
+                kind: Some(CodeActionKind::QUICKFIX),
+                is_preferred: Some(only_one_method),
```

Comment:
maybe we should make the most matching method (float returned from `jaro_winkler`) preferred in case of multiple found?

Reply:
we can do that 👍